projects/demo/src/modules/components/tag/examples/4/index.ts

```diff
@@ -0,0 +1,22 @@
+import {Component} from '@angular/core';
+import {changeDetection} from '@demo/emulate/change-detection';
+import {encapsulation} from '@demo/emulate/encapsulation';
+
+@Component({
+    selector: 'tui-tag-example-4',
+    templateUrl: './index.html',
+    styleUrls: ['./index.less'],
+    changeDetection,
+    encapsulation,
+})
+export class TuiTagExample4 {
+    readonly tags = ['Taiga UI', 'is an open-source library', 'for awesome people'];
+
+    handleTagEdited(newTagCaption: string, tagIndex: number) {
+        if (newTagCaption) {
+            this.tags.splice(tagIndex, 1, newTagCaption);
```

Comment:
![image](https://user-images.githubusercontent.com/11832552/150934479-8f7ba636-821e-49bc-95ee-0875a00740fc.png)

Reply:
@vakhramoff Yes, the splice() method changes the original array and mutation is bad practices

Reply:
![happy-sad-trolling-trollface-2ULn](https://user-images.githubusercontent.com/17404191/150938954-0dd5d1a8-8ace-49f7-8469-93f1c960263d.jpg)